mbed-client-libservice/platform/arm_hal_interrupt.h

```diff
@@ -30,6 +30,19 @@ extern void platform_enter_critical(void);
  */
 extern void platform_exit_critical(void);
 
+/**
+ * \brief This function increments the disable IRQ counter.
+ * The function must be called inside interrupt routines
+ * after platform_enter_critical() has been called.
```

Comment:
More for the comment:

This routine may not be need to do anything on some platforms, but requiring the call to be made before using platform_enter_critical() in an interrupt can simplify some implementations of platform_enter/exit_critical().